pkg/diffs/diffs.go

```diff
@@ -175,3 +175,35 @@ func GetPresubmitsForCiopConfigs(prowConfig *prowconfig.Config, ciopConfigs conf
 
 	return ret
 }
+
+// GetPresubmitsForClusterProfiles returns a filtered list of jobs from the
+// Prow configuration, with only presubmits that use certain cluster profiles.
+func GetPresubmitsForClusterProfiles(prowConfig *prowconfig.Config, profiles []config.ClusterProfile, logger *logrus.Entry) config.Presubmits {
+	ret := config.Presubmits{}
+	for repo, jobs := range prowConfig.JobConfig.Presubmits {
+		for _, job := range jobs {
+			if job.Agent != string(pjapi.KubernetesAgent) {
+				continue
+			}
+		top:
+			for _, v := range job.Spec.Volumes {
+				if v.Name != "cluster-profile" || v.Projected == nil {
+					continue
+				}
+				for _, s := range v.Projected.Sources {
+					if s.ConfigMap == nil {
+						continue
+					}
+					n := strings.TrimPrefix(s.ConfigMap.Name, "cluster-profile-")
+					for _, p := range profiles {
+						if n == p.Name {
+							ret.Add(repo, job)
+							break top
```

Comment:
This is not necessary

Reply:
Not strictly necessary, no, but why bother going through the rest of the iterations if we already know the answer?

Reply:
you will skip only `			if job.Agent != string(pjapi.KubernetesAgent) {
				continue
			}` right?

Reply:
No, `break top` will break from all `for` loops as soon as we find that the job matches our search criteria. Removing the break would cause it to unnecessarily look at all the remaing volumes, sources, and profiles. It would also add duplicates in the (unlikely) case where a job uses the profile more than once or more than one profile.

Reply:
All `for` loops under the `top` label, that is.

Reply:
Named breaks are a code smell -- try to break out the inner loops into functions and keep the indentation low, helps to test as well

Reply:
Generally, I agree, but this is the dumbest of loops. Extracted the relevant part to a local lambda function.